hazelcast/src/main/java/com/hazelcast/util/FutureUtil.java

```diff
@@ -276,6 +278,33 @@ public void handleException(Throwable throwable) {
     }
 
     @PrivateApi
+    public static void waitForever(Collection<Future> futuresToWaitFor, ExceptionHandler exceptionHandler) {
+        Collection<Future> futures = new ArrayList<Future>(futuresToWaitFor);
+        while (true) {
+            Iterator<Future> it = futures.iterator();
+            while (it.hasNext()) {
+                Future future = it.next();
+                try {
+                    future.get();
+                } catch (Exception e) {
```

Comment:
I followed the convention from returnWithDeadline:
https://github.com/tombujok/hazelcast/blob/7ff4f9d8801de72fcfcd497b117589bdaadc7e8d/hazelcast/src/main/java/com/hazelcast/util/FutureUtil.java#L273

Don't know if catching Throwable would be necessarily a good thing here?